util/src/hash.rs

```diff
@@ -539,6 +540,36 @@ impl_hash!(H520, 65);
 impl_hash!(H1024, 128);
 impl_hash!(H2048, 256);
 
+// Specialized HashMap and HashSet
+
+/// Hasher that just takes 8 bytes of the provided value.
+pub struct PlainHasher(u64);
+
+impl Default for PlainHasher {
+	#[inline]
+	fn default() -> PlainHasher {
+		PlainHasher(0)
+	}
+}
+
+impl Hasher for PlainHasher {
+	#[inline]
+	fn finish(&self) -> u64 {
+		self.0
+	}
+
+	#[inline]
+	fn write(&mut self, bytes: &[u8]) {
+		debug_assert!(bytes.len() == 32);
+		self.0 = unsafe { *::std::mem::transmute::<&u8, &u64>(bytes.get_unchecked(0)) };
```

Comment:
you aren't guaranteeing that the byte pointer has the right alignment for a `u64`. this is unsafe.
create a `[u8; 8]` and then transmute that.